### ISSUE TITLE
PCA with Automatic Relevance Determination (ARD)

### ISSUE THREAD

**Issue by minimav** [blei-lab/edward#498]
Hi,

I'm trying to implement a couple of models described in [this](https://arxiv.org/pdf/1603.00788.pdf) paper, specifically F6 (PCA with ARD) and F7 (Supervised PCA with ARD) in the appendix.

Based on the existing [PCA example](https://github.com/blei-lab/edward/blob/master/examples/probabilistic_pca.py), here is my attempt at the former:

```
# N number of data points
# D data dimensionality

# latent variables/features we want to learn
z = Normal(mu=tf.zeros([N, D]), sigma=tf.ones([N, D]))

# noise in the data
sigma = InverseGamma(alpha=1.0, beta=1.0)

# prior that forces some principal components to be omitted
alpha = InverseGamma(alpha=tf.ones(D), beta=tf.ones(D))

# principal components
w = Normal(mu=tf.zeros([D, D]), sigma=sigma * alpha)

# reconstruction of true data from latent features
# good parameters increase likelihood of recreating true data from latent features
x = Normal(mu=tf.matmul(w, z, transpose_b=True), sigma=sigma * tf.ones([D, N]))

qw = Normal(
    mu=tf.Variable(tf.random_normal([D, D])),
    sigma=tf.nn.softplus(tf.Variable(tf.random_normal([D, D])))
)
qz = Normal(
    mu=tf.Variable(tf.random_normal([N, D])),
    sigma=tf.nn.softplus(tf.Variable(tf.random_normal([N, D])))
)

inference = ed.KLqp({w: qw, z: qz}, data={x: x_train})
```

I tried running this on a toy dataset from the PCA example (N=5000, D=2) but the loss I got jumped around so much that I think I messed something up:

```
init = tf.global_variables_initializer()
inference.run(n_iter=1000, n_print=100, n_samples=10)
...
Iteration    1 [  0%]: Loss = 46800.465
Iteration  100 [ 10%]: Loss = 25299.512
Iteration  200 [ 20%]: Loss = 39365.473
Iteration  300 [ 30%]: Loss = 21268.459
Iteration  400 [ 40%]: Loss = 53026.758
Iteration  500 [ 50%]: Loss = 23241.273
Iteration  600 [ 60%]: Loss = 43016.266
Iteration  700 [ 70%]: Loss = 28511.955
Iteration  800 [ 80%]: Loss = 20357.977
Iteration  900 [ 90%]: Loss = 44409.664
Iteration 1000 [100%]: Loss = 32869.477
```

If I change the definition of `w` to 

```
w = Normal(mu=tf.zeros([D, D]), sigma=sigma * tf.reshape(tf.tile(alpha, [D]), [D, D]))
```
which is the only other option that I thought might work, I sometimes get a better loss but it still jumps around an awful lot (more so than above) and so it is not clear to me that my implementation is correct. 

Thank you

**Comment by pwl** [comment]
I've actually done the same thing.  You might want to check out [this](https://gist.github.com/pwl/2f3c3e240b477eac9a37b06791b2a659) gist.

**Comment by pwl** [comment]
By the way, @dustinvtran would it make sense to add this notebook to examples in the repository?

**Comment by minimav** [comment]
Thanks @pwl, that's great.

**Comment by dustinvtran** [comment]
Yes, that would be amazing. We're always looking to up our game re:notebooks. :)
